java/impl/aws/src/main/java/co/worklytics/psoxy/aws/ParameterStoreConfigService.java

```diff
@@ -0,0 +1,60 @@
+package co.worklytics.psoxy.aws;
+
+import co.worklytics.psoxy.gateway.ConfigService;
+import lombok.AllArgsConstructor;
+import lombok.NoArgsConstructor;
+import lombok.extern.java.Log;
+import software.amazon.awssdk.services.ssm.SsmClient;
+
+import software.amazon.awssdk.services.ssm.model.*;
+
+import javax.inject.Inject;
+import java.util.Optional;
+import java.util.logging.Level;
+
+/**
+ * implementation of ConfigService backed by AWS Systems Manager Parameter Store
+ *
+ * https://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-parameter-store.html
+ *
+ */
+//TODO: AssistedFactory??
+//@NoArgsConstructor(onConstructor_ = @Inject)
+// IDE accepts this, but mvn compile complains --> badly linked lombok??
+//[ERROR] /Users/erik/code/psoxy/java/impl/aws/src/main/java/co/worklytics/psoxy/aws/ParameterStoreConfigService.java:[18,20] cannot find symbol
+//[ERROR]   symbol:   method onConstructor_()
+//[ERROR]   location: @interface lombok.NoArgsConstructo
+@Log
+@AllArgsConstructor
+public class ParameterStoreConfigService implements ConfigService {
+
+
+    String namespace;
+
+    @Inject SsmClient client;
+
+    //TODO: add caching?? guava?? apache commons??
```

Comment:
I'd love to work on all kind of improvements here.

Reply:
yeah, there's a task for you